netket/experimental/operator/_particle_number_conserving_fermionic/_operators.py

```diff
@@ -0,0 +1,341 @@
+
+    def get_conn_padded(self, x):
+        return get_conn_padded_pnc(self._operator_data, x, self._hilbert.n_fermions)
+
+    @property
+    @jax.jit
+    def max_conn_size(self):
+        x = jax.ShapeDtypeStruct((1, self._hilbert.size), dtype=jnp.uint8)
+        _, mels = jax.eval_shape(self.get_conn_padded, x)
+        return mels.shape[-1]
+
+    @property
+    def dtype(self):
+        return NotImplemented
+        # return list(self._operator_data.values())[0][2].dtype
+
+    @property
+    def is_hermitian(self):
+        # TODO more efficient implementation
+        return self.to_fermiop().is_hermitian
+
+    @classmethod
+    def _from_coords_data_normal_order(cls, hilbert, coords_data_dict, **kwargs):
+        assert isinstance(hilbert, SpinOrbitalFermions)
+        assert hilbert.n_fermions is not None
+        n_orbitals = hilbert.n_orbitals * hilbert.n_spin_subsectors
+        data = prepare_operator_data_from_coords_data_dict(
+            coords_data_dict, n_orbitals, **kwargs
+        )
+        return cls(hilbert, data)
+
+    @classmethod
+    def _from_sparse_arrays_normal_order(cls, hilbert, operators, **kwargs):
+        terms = sparse_arrays_to_coords_data_dict(collect_ops(operators))
+
+        for k, v in terms.items():
+            if k <= 2:
+                pass
+            idx = v[0]
+            idx_create = idx[:, : idx.shape[1] // 2]
+            idx_destroy = idx[:, idx.shape[1] // 2 :]
+            for idx_arr in idx_destroy, idx_create:
+                if (jnp.diff(idx_arr) > 0).any():
+                    raise ValueError("Input arrays are not in normal order")
+
+        return cls._from_coords_data_normal_order(hilbert, terms, **kwargs)
+
+    @classmethod
+    def from_sparse_arrays(cls, hilbert, operators, **kwargs):
+        # daggers on the left, but not necessarily desc order
+        ops = collect_ops(operators)
+        cutoff = kwargs.get("cutoff", 0)
+        ops = jax.tree_util.tree_map(partial(to_desc_order_sparse, cutoff=cutoff), ops)
+        terms = sparse_arrays_to_coords_data_dict(ops)
+        return cls._from_coords_data_normal_order(hilbert, terms, **kwargs)
+
+    @classmethod
+    def from_fermiop(cls, ha, **kwargs):
+        # ha = ha.to_normal_order()
+        t = fermiop_terms_to_sites_daggers_weights(ha.terms, ha.weights)
+        t = to_normal_order(t)
+        terms = {k: (v[0], v[2]) for k, v in t.items()}  # drop daggers
+        return cls._from_coords_data_normal_order(ha.hilbert, terms, **kwargs)
+
+    def to_fermiop(self, cls=FermionOperator2ndJax):
+        terms = []
+        weights = []
+        for d in self._operator_data:
+            for k, v in d.items():
+                t, w = to_fermiop_helper(*v)
+                terms = terms + t.tolist()
+                weights = weights + w.tolist()
+        return cls(self._hilbert, terms, weights)
+
+    @classmethod
+    def from_pyscf_molecule(cls, mol, mo_coeff, cutoff=1e-11, **kwargs):
```

Comment:
docstring missing.
Would be nice to add the type annotation to `mol` as a string.